File: scala/1929_Concatenation_of_Array.scala
object Solution {
    def getConcatenation(nums: Array[Int]): Array[Int] = {
        var concatArray = new Array[Int](2 * nums.length)
        var k = 0
        for (i <- 0 to (2 * (nums.length) - 1)) {
            if (i < nums.length) {
                k = i
            } else {
                k = i - nums.length
            }
            concatArray(i) = nums(k)
        }
        return concatArray
    }
}

File: rust/1929_Concatenation_of_Array.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let mut concat: Vec<i32> = Vec::new();
        let mut k: usize = 0;
        for i in 0..(2 * nums.len()) {
            if (i < nums.len()) {
                k = i;
            } else {
                k = i - nums.len();
            }
            concat.push(nums[k]);
        }  
        return concat;
    }
}
